cmake/o3deConfigVersion.cmake

```diff
@@ -63,6 +63,25 @@ set(PACKAGE_VERSION ${engine_version})
 # Get the project.json 'engine' field
 string(JSON project_engine ERROR_VARIABLE json_error GET ${o3de_project_json} engine)
 if(json_error OR NOT project_engine)
+    if(NOT project_engine)
+        # Check if the project folder is a subdirectory of this engine and would
+        # be found using scan up logic 
+        cmake_path(GET CMAKE_CURRENT_SOURCE_DIR PARENT_PATH candidate_engine_path)
+        set(reached_root_dir FALSE)
+        while(NOT reached_root_dir)
+            cmake_path(COMPARE "${this_engine_path}" EQUAL "${candidate_engine_path}" is_same_path)
+            if(is_same_path)
+                message(VERBOSE "The project folder is a subdirectory of this engine.")
+                set(PACKAGE_VERSION_COMPATIBLE TRUE)
+                return()
+            else()
+                cmake_path(GET candidate_engine_path PARENT_PATH parent_path)
+                cmake_path(COMPARE "${candidate_engine_path}" EQUAL "${parent_path}" reached_root_dir)
+                set(candidate_engine_path ${parent_path})
+            endif()
+        endwhile()
```

Comment:
Instead of looping, the [cmake_path(IS_PREFIX](https://cmake.org/cmake/help/latest/command/cmake_path.html?highlight=cmake_path#query) function should be used here.

Reply:
nice! great suggestion!

Reply:
Nevermind, it looks like you are trying to locate the engine, not check if the project is a subdirectory

Reply:
I was trying to see if the project is a subdirectory of the engine, IS_PREFIX is much simpler

Reply:
Sounds good.